napari/_qt/widgets/qt_viewer_status_bar.py

```diff
@@ -129,3 +107,120 @@ def _toggle_activity_dock(self, visible: Optional[bool] = None):
         else:
             par._activity_dialog.hide()
             self._activity_item._activityBtn.setArrowType(Qt.ArrowType.UpArrow)
+
+
+class StatusBarWidget(QWidget):
+    def __init__(
+        self,
+        status_label: QLabel,
+        layer_label: QLabel,
+        source_label: QLabel,
+        plugin_label: QLabel,
+        coordinates_label: QLabel,
+        help_label: QLabel,
+        parent: QWidget = None,
+    ):
+        super().__init__(parent=parent)
+        self._status_label = status_label
+        self._layer_label = layer_label
+        self._source_label = source_label
+        self._plugin_label = plugin_label
+        self._coordinates_label = coordinates_label
+        self._help_label = help_label
+
+        self._status_label.setParent(self)
+        self._layer_label.setParent(self)
+        self._source_label.setParent(self)
+        self._plugin_label.setParent(self)
+        self._coordinates_label.setParent(self)
+        self._help_label.setParent(self)
+
+    def resizeEvent(self, event: QResizeEvent) -> None:
+        super().resizeEvent(event)
+        self.do_layout()
+
+    def event(self, event: QEvent) -> bool:
+        if event.type() == QEvent.Type.LayoutRequest:
+            self.do_layout()
+        return super().event(event)
+
+    @staticmethod
+    def _calc_width(fm: QFontMetrics, label: QLabel) -> int:
+        # magical nuber +2 is from superqt code
+        # magical number +12 is from experiments
+        # Adding this values is required to avoid the text to be elided
+        # if there is enough space to show it.
+        return (
+            (
+                fm.boundingRect(label.text()).width()
+                + label.margin() * 2
+                + 2
+                + 12
+            )
+            if label.isVisible()
+            else 0
+        )
+
+    def do_layout(self):
+        width = self.width()
+        height = self.height()
+
+        fm = QFontMetrics(self._status_label.font())
+
+        status_width = self._calc_width(fm, self._status_label)
+        layer_width = self._calc_width(fm, self._layer_label)
+        source_width = self._calc_width(fm, self._source_label)
+        plugin_width = self._calc_width(fm, self._plugin_label)
+        coordinates_width = self._calc_width(fm, self._coordinates_label)
+
+        base_width = (
+            status_width
+            + layer_width
+            + source_width
+            + plugin_width
+            + coordinates_width
+        )
+
+        help_width = max(0, width - base_width)
+
+        if coordinates_width:
+            help_width = 0
```

Comment:
@jni @brisvag 
These two lines remove help text when mouse is over canvas (as there is non empty coordinate text) 

Could you test it? Maybe help text should be shown if there are only coordinates but no properties?

Reply:
🔥 
I love it. Works great.

Reply:
> Maybe help text should be shown if there are only coordinates but no properties?

Either way is fine for me.

Reply:
I think consistency is better. Keep it simple:
- Mouse in canvas: data/properties info.
- Mouse in layers/tools: tools help.